docs/object-capabilities/derived-authority.md

```diff
@@ -0,0 +1,109 @@
+Instead of allowing unfettered access to the outside world, Pony enforces the use of this authority.
+
+Here is a simple program that connects to example.com via TCP on port 80 and quits:
+
+```pony
+use "net"
+
+class MyTCPConnectionNotify is TCPConnectionNotify
+  let _out: OutStream
+
+  new iso create(out: OutStream) =>
+    _out = out
+
+  fun ref connected(conn: TCPConnection ref) =>
+    _out.print("connected")
+    conn.close()
+
+  fun ref connect_failed(conn: TCPConnection ref) =>
+    _out.print("connect_failed")
+
+actor Connect
+  new create(out: OutStream, auth: AmbientAuth) =>
+    TCPConnection(auth, MyTCPConnectionNotify(out), "example.com", "80")
+
+actor Main
+  new create(env: Env) =>
+    try Connect(env.out, env.root as AmbientAuth) end
+```
+
+The Main actor authorizes the Connect actor by passing the ambient authority token on to it. This token is not forgeable since the `AmbientAuth` constructor is private—only the runtime glue code
+can create the one and only instance.
+
+The Connect actor uses this authority when it creates a TCP connection:
+
+```pony
+TCPConnection(auth, MyTCPConnectionNotify(out), "example.com", "80")
+```
+
+The TCPConnection requires an authority as first parameter, and since the compiler checks that
+the correct type was passed, this guarantees that a TCPConnection can only be created by an
+actor holding the required authorization. If you look into the implementation of the `TCPConnection`
+constructor, you will notice that the authorization is not even used beyond the declaration of the
+parameter.
+
+## Restrict, then delegate your authority
+
+In order to handle our own code and that of others more safely, and also to understand our code better,
+we want to split up the authority, and only grant the particular authority a piece of code actually
+requires.
+
+Imagine we don't trust the `Connect` actor, so we don't want to provide it with more authority
+than needed. For example, there is no point in granting it filesystem access.
+
+The first parameter of the `TCPConnection` constructor has the type
+
+```pony
+type TCPConnectionAuth is (AmbientAuth | NetAuth | TCPAuth | TCPConnectAuth)
+```
+
+This looks like a nice hierarchy of authority, in order of decreasing rights. We can pass
+a reduced authority to our `Connect` actor like this:
+
+```pony
+actor Connect
+  new create(out: OutStream, auth: TCPConnectionAuth) =>
+    TCPConnection(auth, MyTCPConnectionNotify(out), "example.com", "80")
+
+actor Main
+  new create(env: Env) =>
+    try Connect(env.out, TCPConnectAuth(env.root as AmbientAuth)) end
+```
+
+Now we are sure it cannot access the filesystem or listen on a TCP or UDP port. This is possible
+because there is a `TCPConnectAuth` constructor which accepts the AmbientAuth.
+
+You can see this pattern over and over in the standard library.
```

Comment:
Remove this please. The subsection does not need to end with a statement about the standard library as it is not about the standard library.